lib/src/package_config_json.dart

```diff
@@ -97,7 +97,7 @@ PackageConfig parsePackageConfigJson(
     return "object";
   }
 
-  T checkType<T>(dynamic value, String name, [String /*?*/ packageName]) {
+  T? checkType<T>(dynamic value, String name, [String? packageName]) {
```

Comment:
This has me thinking that `onError` should have a return type of `Never`? That would likely be breaking... and it would mean the user is still required to throw _something_ from `onError`, which may or may not be desirable.

Reply:
The `onError` is deliberately not required to throw. That allows the front-end to accumulate multiple errors instead of just bailing out on the first one.